### ISSUE TITLE
feat: implement single-h1 rule (MD025)

### ISSUE THREAD

**Issue by shinagawa-web** [shinagawa-web/gomarklint#90]
## Summary

Implement the `single-h1` rule (equivalent to markdownlint MD025) that flags Markdown files containing more than one H1 heading (`#`).

## Background

Tracked in #76 as Priority 1. A document should have exactly one top-level heading to clearly communicate its title and maintain a clean document hierarchy.

## Behavior

Flag every H1 heading after the first one in the file.

**Violation example:**

```markdown
# First Title

Some content.

# Second Title

More content.
```

Expected error:

```
example.md:5: Multiple H1 headings found; only one H1 is allowed per file
```

**Valid example:**

```markdown
# Only Title

## Section One

## Section Two
```

## Scope

- Only `#` (ATX-style) headings are checked — setext-style H1 (`===` underline) is out of scope for now
- H1 headings inside fenced code blocks are ignored
- The rule is enabled by default with `"error"` severity

## Files to Create

- `internal/rule/single_h1.go` — rule function `CheckSingleH1(filename string, lines []string, offset int) []LintError`
- `internal/rule/single_h1_test.go` — unit tests
- `e2e/fixtures/single_h1_violation.md` — fixture with multiple H1s
- `e2e/fixtures/single_h1_valid.md` — fixture with a single H1

## Files to Modify

- `internal/config/config.go` — add to `Default()` and `DefaultConfigJSON`
- `internal/linter/linter.go` — register in `collectErrors()`
- `e2e/e2e_test.go` — add E2E test cases (violation + valid), update file/violation counts if needed

## Implementation Checklist

Follow `.claude/rule-implementation-checklist.md`:

- [ ] Rule function returns `LintError` without setting `Severity` (set by `linter.withSeverity()`)
- [ ] At least one test case passes `offset > 0` to verify line numbers shift correctly
- [ ] Table-driven tests; `Severity` field absent from expected `LintError`
- [ ] E2E covers both success and failure cases
- [ ] `go test ./... -skip TestE2E` passes with 100% coverage
- [ ] No deep if nesting — use early `continue`

## Out of Scope

- Setext-style H1 detection (separate rule or future enhancement)
- `front_matter_title` option (consider frontmatter `title` as an implicit H1) — can be added later as a config option
